File: scratch/inspect_toml_parse.rs
use std::fs;
use simurom_schema::character::CharacterSpec;

fn main() {
    let content = fs::read_to_string("assets/mini_game/characters/blonde.toml").unwrap();
    let spec: CharacterSpec = toml::from_str(&content).unwrap();
    println!("Number of segments parsed: {}", spec.character.segments.len());
    for (i, seg) in spec.character.segments.iter().enumerate() {
        println!("Segment {}: id={}, sprite={}, layer_offset={}", i, seg.id, seg.sprite, seg.layer_offset);
        if let Some(ref blink) = seg.blink {
            println!("  Blink config found!");
            println!("  Blink frames count: {}", blink.blink_frames.as_ref().map(|f| f.len()).unwrap_or(0));
        }
    }
}

File: tools/character-assembler/inspect_toml_parse.py
import tomllib

def main():
    with open("assets/mini_game/characters/blonde.toml", "rb") as f:
        data = tomllib.load(f)
    character = data.get("character", {})
    segments = character.get("segments", [])
    print("Number of segments parsed in TOML:", len(segments))
    for i, seg in enumerate(segments):
        print(f"Segment {i}: id={seg.get('id')}, sprite={seg.get('sprite')}, layer_offset={seg.get('layer_offset')}")
        if "blink" in seg:
            print("  Blink config found!")
            print("  Blink frames count:", len(seg["blink"].get("blink_frames", [])))

if __name__ == "__main__":
    main()
